pkg/config/nodetreemodel/getter.go

```diff
@@ -0,0 +1,234 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package nodetreemodel
+
+import (
+	"maps"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/mohae/deepcopy"
+	"github.com/spf13/cast"
+	"golang.org/x/exp/slices"
+)
+
+// GetKnownKeysLowercased returns all the keys that meet at least one of these criteria:
+// 1) have a default, 2) have an environment variable binded or 3) have been SetKnown()
+// Note that it returns the keys lowercased.
+func (c *ntmConfig) GetKnownKeysLowercased() map[string]interface{} {
```

Comment:
It seems like the return type will always be `map[string]struct{}`, maybe change that to make this function easier to understand? As far as I can tell, the only caller is https://github.com/DataDog/datadog-agent/blob/043922b67eb043ad69e26bb889890d2bd8a95f5d/pkg/config/setup/config.go#L1777 and it's just using this map to do quick lookups.

Reply:
Yes, but we're force to match the same type used by viper which uses interface{} internally. I'll add a comment to remove this along viper.

In all cases, this method will be deleted once viper is removed since we'll know the unknown key directly when loading the YAML.